core/block.js

```diff
@@ -1078,7 +1078,7 @@ Blockly.Block.prototype.updateVarName = function(variable) {
     for (var j = 0, field; field = input.fieldRow[j]; j++) {
       if (field.referencesVariables() &&
           variable.getId() == field.getValue()) {
-        field.setText(variable.name);
+        field.updateVariableName();
```

Comment:
BTW would you rather this be updateVariableName or updateVarName? There are some places in the code base where it is spelled out explicitly, and other places where it is abbreviated (e.g. the function that makes this call).

Reply:
See comment below.